great_expectations/datasource/fluent/pandas_datasource.py

```diff
@@ -616,9 +658,9 @@ def add_dataframe_asset(
         """
         asset: DataFrameAsset = DataFrameAsset(
             name=name,
-            dataframe=dataframe,
             batch_metadata=batch_metadata or {},
         )
+        asset.dataframe = dataframe
```

Comment:
@Kilo59 We are still doing it now (but deprecating).  The goal is to avoid any possible confusion for DataFrames -- ephemeral data references: the add_asset for ephemeral data references accepts only (serializable) metadata.  The only place to specify the (non-serializable) data reference is in `.read_dataframe()` (Pandas only for now) or `.build_batch_request()`.  This will finally make the API for ephemeral data precise and interpretable.  Thanks.